homeassistant/components/tuya/config_flow.py

```diff
@@ -0,0 +1,140 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+# Project Type
+DATA_SCHEMA_PROJECT_TYPE = vol.Schema(
+    {vol.Required(CONF_PROJECT_TYPE, default=0): vol.In(TUYA_PROJECT_TYPE)}
+)
+
+# INDUSTY_SOLUTIONS Schema
+DATA_SCHEMA_INDUSTRY_SOLUTIONS = vol.Schema(
+    {
+        vol.Required(CONF_ENDPOINT): vol.In(TUYA_ENDPOINT),
+        vol.Required(CONF_ACCESS_ID): str,
+        vol.Required(CONF_ACCESS_SECRET): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+# SMART_HOME Schema
+DATA_SCHEMA_SMART_HOME = vol.Schema(
+    {
+        vol.Required(CONF_ACCESS_ID): str,
+        vol.Required(CONF_ACCESS_SECRET): str,
+        vol.Required(CONF_APP_TYPE): vol.In(TUYA_APP_TYPE),
+        vol.Required(CONF_COUNTRY_CODE): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class TuyaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Tuya Config Flow."""
+
+    def __init__(self) -> None:
+        """Init tuya config flow."""
+        super().__init__()
+        self.conf_project_type = None
+        self.project_type = ProjectType.SMART_HOME
+
+    @staticmethod
+    def _try_login(user_input):
+        project_type = ProjectType(user_input[CONF_PROJECT_TYPE])
+        api = TuyaOpenAPI(
+            user_input[CONF_ENDPOINT]
+            if project_type == ProjectType.INDUSTY_SOLUTIONS
+            else "",
+            user_input[CONF_ACCESS_ID],
+            user_input[CONF_ACCESS_SECRET],
+            project_type,
+        )
+        api.set_dev_channel("hass")
+
+        if project_type == ProjectType.INDUSTY_SOLUTIONS:
+            response = api.login(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+        else:
+            api.endpoint = TUYA_ENDPOINT_BASE
+            response = api.login(
+                user_input[CONF_USERNAME],
+                user_input[CONF_PASSWORD],
+                user_input[CONF_COUNTRY_CODE],
+                user_input[CONF_APP_TYPE],
+            )
+            if response.get("success", False):
+                api.endpoint = api.token_info.platform_url
+                user_input[CONF_ENDPOINT] = api.token_info.platform_url
+
+        _LOGGER.debug("TuyaConfigFlow._try_login finish, response:, %s", response)
+        return response
+
+    async def async_step_user(self, user_input=None):
+        """Step user."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA_PROJECT_TYPE
+            )
+
+        self.conf_project_type = user_input[CONF_PROJECT_TYPE]
+        self.project_type = ProjectType(self.conf_project_type)
+
+        return await self.async_step_login(user_input)
+
+    async def async_step_login(self, user_input=None):
+        """Step login."""
+        errors = {}
+        if CONF_USERNAME in user_input:
+            if self.conf_project_type is not None:
```

Comment:
When we get here we know that this is not None. We can just assert that here to satisfy mypy.